sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FieldAccessDescriptor.java

```diff
@@ -513,4 +549,22 @@ private static void validateFieldDescriptor(Schema schema, FieldDescriptor field
       }
     }
   }
+
+  @Override
+  public String toString() {
+    if (getAllFields()) {
+      return "*";
+    }
+
+    List<String> singleSelectors =
+        getFieldsAccessed().stream()
+            .map(FieldDescriptor::getFieldName)
+            .collect(Collectors.toList());
+    List<String> nestedSelectors =
+        getNestedFieldsAccessed().entrySet().stream()
+            .map(e -> e.getKey().getFieldName() + "." + e.getValue().toString())
```

Comment:
Current if you have a.b and a.c, this with return a.b, a.c.
We could make it return a.[b,c], but it might get confusing for multiple nested descriptors? For example then a.b.c.d, a.b.c.e, a.c.f, a.g 

will be printed as  a.[b.[c.[d, e] , f], g]

Where it's probably more readable as simply a.b.c.d, a.b.c.e, a.c.f, a.g